### ISSUE TITLE
switch case unreachable code checking when use string literals

### ISSUE THREAD

**Issue by stepancar** [Microsoft/TypeScript#7745]
Hello! See example
``` javascript
type lang = "en" | "ru";
function getDict(lng: lang): any {
    switch (lng) {
        case 'ru':
            return ru;
        case 'stringNotInLangType':
            return en;
    }
}
```
compiler should throw error, cause 
``` javascript
 case 'stringNotInLangType':
            return en;
```
is unreachable

**Comment by mhegazy** [comment]
duplicate of https://github.com/Microsoft/TypeScript/issues/6149
